src/main/java/org/jabref/logic/importer/fetcher/oaDOI.java

```diff
@@ -0,0 +1,58 @@
+package org.jabref.logic.importer.fetcher;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.jabref.logic.importer.FulltextFetcher;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.FieldName;
+import org.jabref.model.entry.identifier.DOI;
+
+import com.mashape.unirest.http.HttpResponse;
+import com.mashape.unirest.http.JsonNode;
+import com.mashape.unirest.http.Unirest;
+import com.mashape.unirest.http.exceptions.UnirestException;
+import org.json.JSONObject;
+
+/**
+ * A fulltext fetcher that uses <a href="https://oadoi.org/">oaDOI</a>.
+ *
+ * @implSpec API is documented at https://oadoi.org/api/v2
+ */
+public class oaDOI implements FulltextFetcher {
```

Comment:
That's not an "ok" name according to the naming conventions for Java classes: they need to start with upper case. Please rename this to `OaDOI` or maybe `OaDoi`. Maybe `OpenAccessDoi` would be even better.

Reply:
Good point, I wasn't sure about the naming conventions and this service seems to be really called `oaDOI`. But `OpenAccessDoi` is good.